src/ast/dml.rs

```diff
@@ -206,11 +206,7 @@ impl Display for CreateTable {
             name = self.name,
         )?;
         if let Some(on_cluster) = &self.on_cluster {
-            write!(
-                f,
-                " ON CLUSTER {}",
-                on_cluster.replace('{', "'{").replace('}', "}'")
```

Comment:
Why did you make this change? I am not familar with the syntax but it seems like a change in behavior. Maybe we could ping the original author to figure out the rationale for adding the extra `'`

Reply:
@alamb Sorry for forgetting to highlight this change while creating PR. From my understanding, it's used to add a single quote for the cluster name if it's a single quote string. And I moved this behavior into the function: `parse_optional_on_cluster`.

This place was introduced in PR #1291, to see if @philipcristiano can help to review and correct me if I was wrong. Thank you!

Reply:
#1291 moved the code. I'm not familiar with the origination of it or why Clickhouse would have the `on_cluster` quoted. 

On a cursory glance it looks as if SQL generation would be missing quotes. This PR appears to testing parsing `ON CLUSTER` with and without quotes but maybe there wasn't a test for outputting SQL that would detect any change in functionality here.

Reply:
@philipcristiano Thanks for your reply. It's already tested in [parse_create_table_on_cluster](https://github.com/sqlparser-rs/sqlparser-rs/blob/028ada8350d3b2ada4aa67f5e828b318565590f2/tests/sqlparser_common.rs#L3505), so I didn't add in this PR?

Reply:
`ON CLUSTER` for the create table was first introduced in PR #527. @andyrichardson would you mind having a look at this PR?

Reply:
seems the old impl was using the replace functionality to manually inject single quotes when when it was time to display - not familiar with clickhouse so unclear why it used curly braces specifically to anchor the replacement, but in any case I think that behavior would technically be incorrect in some cases like e.g. if clickhouse identifiers allowed curly braces.

With the new behavior, we seem to workaround the issue by instead injecting the single quote during parsing. I wonder if, for correctness and to avoid similar confusion later on, we can instead move away from the current String representation of `on_cluster`, since that the value can be more than a string (e.g. it doesn't look like we'd handle quoted identifiers properly today)? Easiest there would probably be to represent it as an Expr which would be over-representing the type but would let us handle more variants if needed later on, if a dedicated type is deemed to not be worth it in this case.

Reply:
Ah right since ident already includes the single quoted strings then we probably dont need the enum right or? thinking should be enough with `pub on_cluster: Option<Ident>`?

Reply:
Yes, `Option<Ident>` is enough for now. I'm wondering if we should add an enum to extend more values and create a unified display for this enum. But I'm good with `Option<Ident>` to avoid over-design for the future.

Reply:
I imagine we can probably go with Ident and switch to an enum in the future if we need to (if that happens the context would be clearer as to what the enum would look like)

Reply:
Sure, will change to `Option<Ident>`.

Reply:
@iffyio Done, pls have a look at the last commit when you get time. Thank you!